docs/software-architecture-and-design.md

```diff
@@ -472,6 +475,30 @@ The most important thing to know about editing the [Visualizer](#visualizer) in
 
 More information about defining custom widgets in [QtCreator](https://doc.qt.io/qtcreator/creator-using-qt-designer.html) can be found [here](https://doc.qt.io/qt-5/designer-using-custom-widgets.html).
 
+## Qt Best Practices
+Although this will focus on Qt-specific examples, these principles generally apply to all GUI design and implementation. They are really just Software Engineering principles applied to GUIs.
+
+### Build A Hierarchy
+Qt is designed to handle hierarchy. It has [an extensive and robust system for maintaining Object Trees and parent/child relationships between components](https://doc.qt.io/qt-5/objecttrees.html). Much like regular code, GUIs should be created in a logical hierarchy to make organization and re-use easier.
+
+Do **not** lay out all your widgets in a single layer like this:
+![Bad Qt Widget Layout](images/qt_layout_bad_example.png)
+
+Make sure to use layouts, and group widgets in a logical way. For example, several widgets that work together to collect a user's mailing address should be grouped. This group may then be part of a larger group of widgets in a popup dialog asking for billing information.
+
+Here is a good example of laying out widgets in a hierarchy:
+![Good Qt Widget Layout](images/qt_layout_good_example.png)
+
+The main point to remember is to use [layouts](https://doc.qt.io/qt-5/layout.html) to group and manage widgets, and to create these groups in a logical way that can be built into a hierarchy. This will make it significantly easier to replace parts of the GUI later, or move components around.
+
+### Create Reusable Widgets
```

Comment:
we should pull this common code out into a common folder, can you make an issue?

Reply:
Yup I'm planning to make a bunch of GUI-related tickets